types/relay.go

```diff
@@ -36,3 +64,42 @@ type Relay struct {
 	CreatedAt                time.Time     `json:"createdAt"`
 	UpdatedAt                time.Time     `json:"updatedAt"`
 }
+
+func (r Relay) Validate() (err error) {
+	structType := reflect.TypeOf(r)
+	structVal := reflect.ValueOf(r)
+	fieldNum := structVal.NumField()
+
+	var isError bool
+
+	for i := 0; i < fieldNum; i++ {
+		field := structVal.Field(i)
+		fieldName := structType.Field(i).Name
+
+		isSet := field.IsValid() && !field.IsZero()
+
+		if isSet {
+			if fieldName == "IsError" {
+				isError = true
+			}
+
+			if shouldBeEmptyField[fieldName] || (!isError && errorField[fieldName]) {
```

Comment:
going along with the previous comment, what if the errorField comes before `isError`

Reply:
It is not unordered